### PR TITLE
MCR-2263 File upload should support derivate_types

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2263).

### CODE REVIEW

File: mycore-webtools/src/main/java/org/mycore/webtools/upload/MCRUploadResource.java [hunk 1]
```diff
@@ -181,18 +192,44 @@ public void commit(@QueryParam("uploadID") String uploadID) {
         MCRFileUploadBucket.releaseBucket(uploadID);
     }
 
+    private List<MCRMetaClassification> getClassifications(String classifications) {
+        final MCRCategoryDAO dao = MCRCategoryDAOFactory.getInstance();
+        final List<MCRCategoryID> categoryIDS = Stream.of(classifications)
+            .filter(Objects::nonNull)
+            .filter(Predicate.not(String::isBlank))
```
Comment (yagee-de) [inline mycore-webtools/src/main/java/org/mycore/webtools/upload/MCRUploadResource.java:199]: seems not necessary

File: mycore-webtools/src/main/java/org/mycore/webtools/upload/MCRUploadResource.java [hunk 2]
```diff
@@ -181,18 +192,44 @@ public void commit(@QueryParam("uploadID") String uploadID) {
...
+            .filter(cv -> cv.contains(":"))
+            .map(classValString -> {
+                final String[] split = classValString.split(":");
+                return new MCRCategoryID(split[0], split[1]);
+            }).collect(Collectors.toList());
+
+        if (!categoryIDS.stream().allMatch(dao::exist)) {
+            final String parsedIDS = categoryIDS.stream().map(Object::toString).collect(Collectors.joining(","));
+            throw new MCRException(String.format(Locale.ROOT, "One of the Categories \"%s\" was not found", parsedIDS));
```
Comment (yagee-de) [inline mycore-webtools/src/main/java/org/mycore/webtools/upload/MCRUploadResource.java:210]: Which one?

**Review by yagee-de — APPROVED** [review]
did only review the java part, which looks ok with minor issues

**Review by eckstueck — APPROVED** [review]
TypeScript part looks good

**Pull request merged**